src/d3d9/d3d9_device.cpp

```diff
@@ -4655,18 +4655,44 @@ namespace dxvk {
     // D3D9 doesn't have the concept of a framebuffer object,
     // so we'll just create a new one every time the render
     // target bindings are updated. Set up the attachments.
+    VkSampleCountFlagBits sampleCount;
+    VkExtent2D extent;
+    if (likely(!m_state.renderTargets[0]->IsNull())) {
+        const DxvkImageCreateInfo& rtImageInfo = m_state.renderTargets[0]->GetCommonTexture()->GetImage()->info();
+        sampleCount   = rtImageInfo.sampleCount;
+        extent.width  = rtImageInfo.extent.width;
+        extent.height = rtImageInfo.extent.height;
+    } else if (likely(m_state.depthStencil != nullptr)) {
+        const DxvkImageCreateInfo& dsImageInfo = m_state.depthStencil->GetCommonTexture()->GetImage()->info();
+        sampleCount   = dsImageInfo.sampleCount;
+        extent.width  = dsImageInfo.extent.width;
+        extent.height = dsImageInfo.extent.height;
+    } else {
+      Logger::err("RT0 and DS unbound");
+      return;
+    }
+
     for (UINT i = 0; i < m_state.renderTargets.size(); i++) {
       if (m_state.renderTargets[i] != nullptr && !m_state.renderTargets[i]->IsNull()) {
+        const DxvkImageCreateInfo& rtImageInfo = m_state.renderTargets[i]->GetCommonTexture()->GetImage()->info();
+        if (unlikely(rtImageInfo.sampleCount != sampleCount
+          || rtImageInfo.extent.width != extent.width
+          || rtImageInfo.extent.height != extent.height))
+          continue;
+
         attachments.color[i] = {
           m_state.renderTargets[i]->GetRenderTargetView(srgb),
           m_state.renderTargets[i]->GetRenderTargetLayout() };
       }
     }
 
-    if (m_state.depthStencil != nullptr) {
-      attachments.depth = {
-        m_state.depthStencil->GetDepthStencilView(),
-        m_state.depthStencil->GetDepthLayout() };
+    if (likely(m_state.depthStencil != nullptr)) {
```

Comment:
I don't think this case should be given higher precedence over the alternative.

Reply:
Do you mean I should get rid of the `likely`? Because there's no way draws without a depth buffer are more common.